app/ydoc-server/src/languageServerSession.ts

```diff
@@ -505,6 +513,7 @@ class ModulePersistence extends ObservableV2<{ removed: () => void }> {
     const newSnapshot = newCode && {
       snapshot: ModulePersistence.encodeCodeSnapshot(newCode),
     }
+    if (newMetadata) newMetadata.snapshot = this.syncedMeta.ide.snapshot
```

Comment:
The #11435 changes the `newMetadata` argument to store the `ide` object:
```diff
-    newMetadata: fileFormat.IdeMetadata['node'] | undefined,
+    newMetadata: fileFormat.IdeMetadata | undefined,
 ```
 
This change https://github.com/enso-org/enso/pull/11435/files#diff-f621895c66ac2e5e220af1fb6ce9cd3694e5f85897f471300ad63a7c9f5bee9aL511-L513 builds the new metadata object without the `snapshot` field if there are no code changes. I.e. it makes the `snapshot` field disappear from the file on metadata changes (moving the node, for example)